dlio_benchmark/checkpointing/tf_checkpointing.py

```diff
@@ -62,35 +58,39 @@ def get_tensor_core(self, length, datatype="int8", randomize=True):
         tf_dtype = get_tf_datatype(datatype)
         if randomize:
             if tf_dtype in [tf.float16, tf.float32, tf.float64, tf.bfloat16]:
-                return tf.random.uniform(shape=(length,), minval=0, maxval=1, dtype=tf_dtype)
+                tensor = tf.random.uniform(shape=(length,), minval=0, maxval=1, dtype=tf_dtype)
             elif tf_dtype == tf.int8:
                 random_tensor = tf.random.uniform(shape=(length,), minval=-128, maxval=128, dtype=tf.int32)
-                return tf.cast(random_tensor, dtype=tf.int8)
+                tensor = tf.cast(random_tensor, dtype=tf.int8)
             elif tf_dtype == tf.uint8:
                 random_tensor = tf.random.uniform(shape=(length,), minval=0, maxval=256, dtype=tf.int32)
-                return tf.cast(random_tensor, dtype=tf.uint8)
+                tensor = tf.cast(random_tensor, dtype=tf.uint8)
             else:
-                 raise Exception(f"Datatype {tf_dtype} cannot be randomized for random tensor generation.")
-        return tf.ones((length), dtype=tf_dtype)
```

Comment:
Why did you change logic?

Reply:
I am not changing the logic. the previous checkpointing implementation of tensorflow did not work as intended. what I meant by not working is: *tensor that is saved into a ckpt file is not loaded from the checkpoint file since it fails at saving the checkpoint.*

i found this issue in the same version of the tensorflow specified in the setup.py/requirements.txt

based on my comment in the PR

> We also have problem saving/loading the checkpoint in tensorflow. tensorflow requires the checkpointing data to be "trackable" (https://www.tensorflow.org/guide/checkpoint#object_tracking), thus using tensor is not enough. I modified the tensor to be `tf.Variable` to make it trackable

so the current code now will change the tensor into `tf.Variable`

maybe @zhenghh04 can chime in?

Reply:
I think the changes make sense.